### DISCUSSION TITLE
Feature Request: Event to override supported sites

### DISCUSSION THREAD

**Discussion by ghost** [craftcms/cms#8704] [Ideas]
### Why do I need this?
I'm currently having some propagation issue's while running my custom data sync. When I built my plugin I only had one site so its not built to work with propagation. I don't need it either which is why I would like to disable it. This unfortunately isn't possible as Craft Commerce does not allow disabling sites.

### Description
An easy yet effective solution would be adding an event which you can use to override the supported sites of an element. I don't see how this could do any harm but I think it would be fairly easy to implement and use. This would allow me to make my Commerce products only support the default site and thereby disabling propagation.

I hope this feature will be added in the near future. Thanks in advance!

**Comment by tim-van-dijkhuizen** [comment]
@brandonkelly What would be a fitting release for such change? I'm currently having to use a forked version so I would like to submit a PR for it. It would be a breaking change, so I'm afraid it won't be merged any time soon.

Edit: I can implement this with a low chance of things breaking and without writing ugly work-arounds. Expect a PR soon!

**Comment by tim-van-dijkhuizen** [comment]
So after a while I realized that this FR is not going to work. I did manage to [implement this FR](https://github.com/TimvanDijkhuizen/cms/tree/supported-sites-event) and it should not be a breaking change. However, I ran into some issue's with sources because those are defined by a static function. Because of this I'm unable to exclude unsupported sites from the sources.

I came to the conclusion that the only good solution would be to implement site settings for Commerce product types. This way I can disable all non-primary sites stopping any propagation. I'm kind of surprised Commerce doesn't have this feature already.

I will create a PR anyways because it might be useful to have this event. I could use it together with the sources event to remove Product support for the non-primary site.
